File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        if (needle.size() > haystack.size()) return -1;
        if (needle.size() == haystack.size()) return (haystack == needle ? 0 : -1);
        int i = 0, j = needle.size(),n = haystack.size();
        for (;i+j-1 < n;i++) {
            if (haystack.substr(i, j) == needle)
                return i;
        }
        return -1;
    }
};

File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.php
class Solution {

    /**
     * @param String $haystack
     * @param String $needle
     * @return Integer
     */
    function strStr($haystack, $needle) {
        $position = strpos($haystack,$needle);
        return (($position === false) ? -1 : $position);
    }
}